### PR TITLE
Rebar3

### PR DESCRIPTION
**Pull request by michalwski**
In this PR:
* use rebar3 to build and test escalus
* update deps and use hex pkgs where possible
* change from `mustache` to `bbmustache` in order to be consistent with MongooseIM's test

### CODE REVIEW

**Comment by michalwski** [comment]
This is tried with MognooseIM here: esl/MongooseIM#2182

File: test/escalus_stanza_SUITE.erl [hunk 1]
```diff
@@ -53,7 +53,7 @@ attribute_as_argument(_) ->
     Attr = {<<"name">>, <<"value">>},
     Example = #xmlel{name = <<"el">>,
                      attrs = [Attr]},
-    ?eq(Example, M:from_template("<el {{attr}}/>", [{attr, Attr}])).
+    ?eq(Example, M:from_template("<el {{{attr}}}/>", [{attr, Attr}])).
```
Comment (erszcz) [inline test/escalus_stanza_SUITE.erl]: What exactly does this change do? We might need to update tests when switching to this new version if this is required.

File: src/escalus_stanza.erl [hunk 2]
```diff
@@ -893,15 +893,20 @@ xml_to_xmlterm(XML) when is_binary(XML) ->
       Snippet :: xml_snippet(),
       Ctx :: context(),
       Text :: string().
+
 render(Snippet, Ctx) ->
-    mustache:render(xml_snippet_to_string(Snippet),
-                    validate_context(Ctx)).
+    ValidateCtx = validate_context(Ctx),
+    ct:pal("WTF: ~p~n~p", [Ctx, ValidateCtx]),
```
Comment (erszcz) [inline src/escalus_stanza.erl]: Call to `ct` outside `escalus_ct`

File: src/escalus_stanza.erl [hunk 3]
```diff
@@ -893,15 +893,20 @@ xml_to_xmlterm(XML) when is_binary(XML) ->
...
+    R = bbmustache:render(xml_snippet_to_binary(Snippet),
+                      ValidateCtx),
+    ct:pal("Render: ~p~n~p", [Snippet, R]),
```
Comment (erszcz) [inline src/escalus_stanza.erl]: Call to `ct` outside `escalus_ct`

**Comment by erszcz** [comment]
Looks good, I'm in favour of finally switching to Rebar3.

File: src/escalus_stanza.erl (hunk 3 above)
Comment (michalwski) [reply src/escalus_stanza.erl]: @erszcz I just removed and the other call to `ct:pal`.

**Pull request merged**